script.module.autocompletion/lib/AutoCompletion.py

```diff
@@ -0,0 +1,234 @@
+        super(BingProvider, self).__init__(*args, **kwargs)
+
+    def fetch_data(self, search_str):
+        url = "query=%s" % (quote_plus(search_str))
+        result = get_JSON_response(url=self.BASE_URL + url,
+                                   headers=HEADERS,
+                                   folder="Bing")
+        if not result:
+            return []
+        else:
+            return result[1]
+
+
+class LocalDictProvider(BaseProvider):
+
+    def __init__(self, *args, **kwargs):
+        super(LocalDictProvider, self).__init__(*args, **kwargs)
+
+    def get_predictions(self, search_str):
+        """
+        get dict list with autocomplete labels from locally saved lists
+        """
+        listitems = []
+        k = search_str.rfind(" ")
+        if k >= 0:
+            search_str = search_str[k + 1:]
+        local = SETTING("autocomplete_lang_local")
+        path = os.path.join(ADDON_PATH, "resources", "data", "common_%s.txt" % (local if local else "en"))
+        with codecs.open(path, encoding="utf8") as f:
+            for line in f.readlines():
+                if not line.startswith(search_str) or len(line) <= 2:
+                    continue
+                li = {"label": line,
+                      "search_string": line}
+                listitems.append(li)
+                if len(listitems) > self.limit:
+                    break
+        return listitems
+
+
+def get_JSON_response(url="", cache_days=7.0, folder=False, headers=False):
+    """
+    get JSON response for *url, makes use of file cache.
+    """
+    now = time.time()
+    hashed_url = hashlib.md5(url.encode()).hexdigest()
+    if folder:
+        cache_path = xbmc.translatePath(os.path.join(ADDON_DATA_PATH, folder))
+    else:
+        cache_path = xbmc.translatePath(os.path.join(ADDON_DATA_PATH))
+    path = os.path.join(cache_path, hashed_url + ".txt")
+    cache_seconds = int(cache_days * 86400.0)
+    if xbmcvfs.exists(path) and ((now - os.path.getmtime(path)) < cache_seconds):
+        results = read_from_file(path)
+        log("loaded file for %s. time: %f" % (url, time.time() - now))
+    else:
+        response = get_http(url, headers)
+        try:
+            results = json.loads(response)
+            log("download %s. time: %f" % (url, time.time() - now))
+            save_to_file(results, hashed_url, cache_path)
+        except Exception as error:
+            log(error)
+            log("Exception: Could not get new JSON data from %s. Tryin to fallback to cache" % url)
+            log(response)
+            if xbmcvfs.exists(path):
+                results = read_from_file(path)
+            else:
+                results = []
+    if results:
+        return results
+    else:
+        return []
+
+
+def get_http(url=None, headers=False):
+    """
+    fetches data from *url, returns it as a string
+    """
+    succeed = 0
+    if not headers:
+        headers = {'User-agent': 'XBMC/16.0 ( phil65@kodi.tv )'}
```

Comment:
Why this headers are any different from the global on the top of the file? Also xbmc 16.0

Reply:
was already there. changed.